### PR TITLE
Refactor Helm chart structures

### PR DESCRIPTION
**Pull request by rinx**
### Description:

Following the upstream k8s deployments.
Helm chart directory structure is refactored.

### Related Issue:

Nothing.

### How Has This Been Tested?:

In my local KinD.

### Environment:

- Golang Version: 1.13.5
- Docker Version: 19.03.5
- Kubernetes Version: 1.16.3
- NGT Version: 1.8.4

### Types of changes:

- [ ] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [ ] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [X] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you successfully ran tests with your changes locally?

### Checklist:

- [X] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [X] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [ ] I have added tests and benchmarks to cover my changes.
- [ ] I have ensure all new and existing tests passed.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: vald/templates/gateway/vald/deployment.yaml [hunk 1]
```diff
@@ -37,6 +37,59 @@ spec:
       labels:
         app: {{ .Values.gateway.name }}
     spec:
+      initContainers:
+      {{- if .Values.gateway.initContainers.waitForManagerCompressor.enabled }}
+      {{- $compressorReadinessPort := default .Values.defaults.server_config.healths.readiness.port .Values.compressor.server_config.healths.readiness.port }}
+      - name: wait-for-manager-compressor
+        image: busybox
```
Comment (kpango) [inline vald/templates/gateway/vald/deployment.yaml]: this should be configurable by values.yaml

File: vald/templates/manager/compressor/deployment.yaml [hunk 2]
```diff
@@ -37,11 +37,24 @@ spec:
       labels:
         app: {{ .Values.compressor.name }}
     spec:
+      initContainers:
+      {{- if .Values.compressor.initContainers.waitForManagerBackup.enabled }}
+      {{- $backupManagerReadinessPort := default .Values.defaults.server_config.healths.readiness.port .Values.backupManager.server_config.healths.readiness.port }}
+      - name: wait-for-manager-backup
+        image: busybox
```
Comment (kpango) [inline vald/templates/manager/compressor/deployment.yaml]: same

**Comment by kpango** [comment]
why redis/agent-ngt's configmap were deleted?

**Comment by rinx** [comment]
> why redis/agent-ngt's configmap were deleted?

just moved.

File: vald/templates/gateway/vald/deployment.yaml [hunk 3]
```diff
@@ -37,6 +37,59 @@ spec:
...
+        command:
+        - /bin/sh
+        - -c
+        - >
+          set -x;
+          while [ $(curl -sw '%{http_code}' "http://{{ .Values.compressor.name }}.{{ .Release.Namespace }}.svc.cluster.local:{{ $compressorReadinessPort }}" -o /dev/null) -ne 200]; do
+            sleep 2;
```
Comment (kpango) [inline vald/templates/gateway/vald/deployment.yaml]: sleed duration should be configurable by values.yaml

**Comment by kpango** [comment]
Others LGTM, please fix some commented code

**Comment by rinx** [comment]
@kpango Thanks for your review.
I've just finished to revise this PR.

**Comment by kpango** [comment]
could you please explain about initContainers, I think we have to wait for redis & cassandra & mysql are ready, but I can't find these initContainer,

**Comment by rinx** [comment]
vald/templates/jobs/db/initializers could be better, however, it will be done later.

File: k8s/agent/configmap.yaml [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+#
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: vald-agent-ngt-config
+data:
+  config.yaml: |
+    ---
+    version: v0.0.0
+    server_config:
```
Comment (kpango) [inline k8s/agent/configmap.yaml]: where is time_zone definition?

**Comment by rinx** [comment]
Recently, k8s dir not updated (because it'll automatically be synchronized once it is merged into master).
Now it has been updated.

**Comment by kpango** [comment]
Okay thanks

**Pull request merged**